### PR TITLE
feat: update components to extend HTML attributes with classNameOverride (part 4)

### PR DESCRIPTION
**Pull request by HeartSquared**
# What

Part 4

Updates following components to extend HTML attributes with `classNameOverride`:

- SplitButton
- Tile
	- InformationTile
	- MultiActionTile
	- TileGrid
- Typography (just updated these to use `OverrideClassName`)
	- Heading
	- Paragraph
- Well

# Why

To prevent us being roadblocks for our consumers

### CODE REVIEW

File: draft-packages/tile/KaizenDraft/Tile/InformationTile.tsx [hunk 1]
```diff
@@ -1,32 +1,10 @@
...
-    titleTag={titleTag}
-    metadata={metadata}
-    information={information}
-    footer={footer}
-    mood={mood}
-  >
-    {children}
-  </GenericTile>
+export const InformationTile: React.VFC<InformationTileProps> = props => (
+  <GenericTile {...props} />
```
Comment (mcwinter07) [inline draft-packages/tile/KaizenDraft/Tile/InformationTile.tsx:7]: look at all that red! #feelsgood

File: draft-packages/tile/KaizenDraft/Tile/components/GenericTile.tsx [hunk 2]
```diff
@@ -1,58 +1,56 @@
...
-  readonly information?: TileInformation | React.ReactNode
-  readonly mood?:
+export interface GenericTileProps
+  extends OverrideClassName<Omit<HTMLAttributes<HTMLDivElement>, "title">> {
+  children?: React.ReactNode
+  title: React.ReactNode
+  titleTag?: AllowedHeadingTags
+  metadata?: string
+  information?: TileInformation | React.ReactNode
+  mood?:
```
Comment (mcwinter07) [inline draft-packages/tile/KaizenDraft/Tile/components/GenericTile.tsx:34]: Now that we've got something like component-base we could potentially house our type for moods type in there. Not necessarily in this fix, but something worth considering 🤔 💭

**Review by mcwinter07 — APPROVED** [review]
There's a food for thought comment about moods in the GenericTile but this all looks hunky dory to me ✅

File: draft-packages/tile/KaizenDraft/Tile/components/GenericTile.tsx (hunk 2 above)
Comment (HeartSquared) [reply draft-packages/tile/KaizenDraft/Tile/components/GenericTile.tsx:34]: Certainly something we could consider later :) The `component-base` is meant to house shared things after all.

**Pull request merged**